platform/fabric/core/generic/vault/vault.go

```diff
@@ -90,7 +91,7 @@ func (p *populator) Populate(rws *vault.ReadWriteSet, rwsetBytes []byte, namespa
 				bn = read.Version.BlockNum
 				txn = read.Version.TxNum
 			}
-			rws.ReadSet.Add(ns, read.Key, bn, txn)
+			rws.ReadSet.Add(ns, read.Key, fver.ToBytes(bn, txn))
```

Comment:
I think the Fabric Platform should provide this function not `core/generic/vault` as this is Fabric specific. Other platforms may implement different formats.